src/apis/group/group.service.js

```diff
@@ -421,43 +418,47 @@ class GroupService {
                 updatedBy: createdGroup.updatedBy
             }
             this.createGroupUserMap(groupUserMap, (userMapped) => {});
-            //mapping bot to the group and creating a new message
-            sequelize
-                .query("select u.id, u.firstname, u.role, u.email, count(gu.userId) from user u LEFT JOIN group_user_map gu on u.id=gu.userId and u.role='BOT' group by u.id order by count(gu.userId) ASC", { type: sequelize.QueryTypes.SELECT })
-                .then((groupUserMaps) => {
-                    var uId;
-                    for (var i = 0; i < groupUserMaps.length; i++) {
-                        if (groupUserMaps[i].role.toLowerCase() == 'bot') {
-                            uId = groupUserMaps[i].id;
-                            break;
-                        } else {
-                            continue;
-                        }
-                    }
-                    var groupUserMapBot = {
-                        groupId: createdGroup.id,
-                        userId: uId,
-                        createdBy: createdGroup.createdBy,
-                        updatedBy: createdGroup.updatedBy
-                    };
-                    groupUserMapDao.insert(groupUserMapBot, (createdGroupUserMap) => {});
-                    var msg = {
-                        receiverId: createdGroup.id,
-                        receiverType: 'group', // group or individual
-                        senderId: uId,
-                        text: 'Welcome to Mesomeds! I am Medroid, your medical assistant. How may I assist you?',
-                        createdTime: Date.now(),
-                        updatedTime: Date.now()
-                    };
-                    messageService.sendMessage(msg, (result) => {});
+            //mapping bot(same bot which is in MedHelp) to the consutation group and create a new message
+            groupUserMapModel.group_user_map.findAll({
+                where: { userId: patientId },
+                order: [
+                    ['createdAt', 'ASC']
+                ],
+                limit: 1
+            }).then((groupUserMap) => {
+                this.getById(groupUserMap[0].groupId, (group) => {
+                    groupUserMapModel.group_user_map.findAll({ where: { groupId: group.id } }).then((groupUserMaps) => {
+                        groupUserMaps.map((groupUserMap) => {
+                            userService.getById(groupUserMap.userId, ((user) => {
+                                if (user.role.toLowerCase() == 'bot') {
+                                    var groupUserMapBot = {
```

Comment:
Send the condition along with the query instead of `if` block. It will return you only one user and that will be bot. Am I right?